src/data/proxy/DDBPreferenceCorrelationGraph.java

```diff
@@ -108,4 +115,18 @@ public Preference getPreference(String id, PreferenceCategory category) {
             return new DDBPreferenceAdapter(item).toObject();
         }
     }
+    
+    private List<Preference> getPreferences(Page<Item, ScanOutcome> page) {
+    	Iterator<Item> iterator = page.iterator();
+    	return Stream.generate(iterator::next).map(item -> new DDBPreferenceAdapter(item).toObject()).collect(Collectors.toList());
+    }
+
+	@Override
+	public Stream<List<Preference>> batchGetPreferences(
+			PreferenceCategory category, int batchSize) {
+		Stream<Page<Item, ScanOutcome>> rawStream = StreamSupport.stream(this.preferenceTable.scan(
+				new ScanFilter(PREFERENCE_ID_ATTRIBUTE).contains(category.toString()))
```

Comment:
I don't really like attaching the assumption that the ID contains the category to this piece of code. Furthermore, it could also yield false positives. For example, say you're looking for preferences in the BOOKS category, and you see an ID of "STORES~~DavesBOOKS" or something similar. I understand that you're sort of limited by what ScanFilter provides, but maybe there's a better way to do this.